styx-common/src/main/java/com/spotify/styx/model/DataEndpoint.java

```diff
@@ -49,15 +49,19 @@
   @JsonProperty
   public abstract Optional<Secret> secret();
 
+  @JsonProperty
+  public abstract Optional<List<String>> resources();
+
   @JsonCreator
   public static DataEndpoint create(
       @JsonProperty("id") String id,
       @JsonProperty("partitioning") Partitioning partitioning,
       @JsonProperty("docker_image") Optional<String> dockerImage,
       @JsonProperty("docker_args") Optional<List<String>> dockerArgs,
-      @JsonProperty("secret") Optional<Secret> secret) {
+      @JsonProperty("secret") Optional<Secret> secret,
+      @JsonProperty("resources") Optional<List<String>> resources) {
```

Comment:
I think jackson can handle parsing a missing key directly into an empty list, without having to wrap it in an optional. For the `resources` field, a missing key and an empty list should have the same meaning.